src/plots/cartesian/axis_defaults.js

```diff
@@ -117,6 +125,38 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, coerce,
         coerce('tickson', ticksonDflt);
     }
 
+    // adjust ticklabelposition to in respect to axis orientation & in-between modes
+    if(containerOut.ticklabelposition) {
+        if(letter === 'x') {
+            containerOut.ticklabelposition =
+            containerOut.ticklabelposition
+                .replace(' top', '')
+                .replace(' bottom', '');
```

Comment:
The pattern we've normally used for this kind of situation is to change the available options based on other factors. See for example what we did with `side`:

https://github.com/plotly/plotly.js/blob/432e448161797aeeffe06a26e4a8ab2ae98524ec/src/plots/cartesian/position_defaults.js#L50-L56

This means we wouldn't try to keep part of the user's intention as you do here (currently "inside" would survive, even if "top" didn't) but I think that's better, plain "inside" will probably look bad when you meant "inside right" but you actually said "inside top".

Reply:
Thanks for the note.
Addressed in https://github.com/plotly/plotly.js/pull/5275/commits/2361958804bebee2c789382367976ad2517f0ca2.
Resolving...